test/app/lib/cinematic/AbstractCommand.spec.js

```diff
@@ -29,23 +41,48 @@ describe('AbstractCommand', () => {
   })
 
   it('run must return a promise', () => {
-    const invalidRunFunctions = [
-      () => undefined,
-      () => true,
-      () => false,
-      () => '',
-      () => 'abc',
-      () => {},
-      () => ({ then: () => { } }), // Fake promise
-      () => Promise.resolve() // non cancellable promise
-    ]
-
-    for (const fn of invalidRunFunctions) {
+    for (const fn of simpleRunFunctions) {
       expect(() => (new PassInRunCommand(fn))[run]()).toThrow()
     }
   })
 })
 
+describe('SyncFunction', () => {
+  it('runs any function passed in', () => {
+    for (const fn of simpleRunFunctions) {
+      expect(() => (new SyncFunction(fn))[run]()).not.toThrow()
```

Comment:
Should we also check if `run` is called here?

Reply:
That's a great additional test. I'll add that! Thanks!